src/server/server.ts

```diff
@@ -7,12 +7,58 @@ namespace ts.server {
     const readline: NodeJS.ReadLine = require("readline");
     const fs: typeof NodeJS.fs = require("fs");
 
+    // TODO: "net" module not defined in local node.d.ts
+    const net: any = require("net");
+
     const rl = readline.createInterface({
         input: process.stdin,
         output: process.stdout,
         terminal: false,
     });
 
+    // Need to write directly to stdout, else rl.write also causes an input "line" event
+    // See https://github.com/joyent/node/issues/4243
+    let writeHost = (data: string) => process.stdout.write(data);
+
+    // Stubs for I/O
+    const onInput = (input: string) => { return; };
+    const onClose = () => { return; };
+
+    // Use a socket for comms if defined
+    const tss_debug: string = process.env["TSS_DEBUG"];

```

Comment:
All these edits to server.ts look like changes I made in my branch to help with debugging Sublime but hadn't pushed to 'master' (and are part of bigger changes on both Sublime and tsserver to use TCP over stdin/stdout if configured to do so via `TSS_DEBUG`). Any idea how the changes in this file got in this code review?